cmd/plume/fcos.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Red Hat Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/spf13/pflag"
+)
+
+var (
+	specPolicy   string
+	specCommitId string
+
+	fcosSpec = fcosChannelSpec{
+		Bucket:  "fcos-builds",
+		Profile: "default",
+		Region:  "us-east-1",
+	}
+)
+
+func AddFcosSpecFlags(flags *pflag.FlagSet) {
+	flags.StringVar(&specPolicy, "policy", "public-read", "Canned ACL policy")
+	flags.StringVar(&specCommitId, "commit-id", "", "OSTree Commit ID")
+}
+
+func FcosChannelSpec() fcosChannelSpec {
```

Comment:
This function is confusing me. So, it returns a statically initialized struct, *but* verifies some runtime vars, some of which don't seem to be defined in this file? Can we move those checks to `runFcosRelease` ?

Reply:
This is how it's implemented for other distros (see CL here: https://github.com/coreos/mantle/blob/master/cmd/plume/containerlinux.go#L350-L411). Part of that reasoning is because we want that check ran for both prerelease & release (which would require duplication rather than handling it in the `channelSpec` functions as well as some runtime vars being required for fetching the correct `channelSpec`).

Given FCOS is pretty divergent WDYT about creating another function specifically for checking these runtime vars and calling it separately inside of `runFcosRelease`?

Reply:
Yup, I like the new approach!